src/lib/addEventListeners.spec.js

```diff
@@ -1,19 +1,221 @@
+      wrapper.id = 'test';
+      const button = document.createElement('button');
+      button.setAttribute('data-htz-dialog-hide', 'test');
+      const dialogState = { wrapperId: wrapper.id, wrapper, hide: sinon.spy() };
+      document.body.appendChild(wrapper);
+      document.body.appendChild(button);
+
+      addEventListeners(dialogState);
+      button.click();
+      expect(dialogState.hide).to.have.been.called();
+
+      // Cleanup
+      wrapper.remove();
+      button.remove();
+    });
+
+    it('Executes hide() when the wrapper is clicked', () => {
+      const wrapper = document.createElement('div');
+      const dialogState = { wrapperId: wrapper.id, wrapper, hide: sinon.spy() };
+      document.body.appendChild(wrapper);
+
+      addEventListeners(dialogState);
+      wrapper.click();
+      expect(dialogState.hide).to.have.been.called();
+
+      // Cleanup
+      wrapper.remove();
+    });
+
+    it('Does not execute hide() when an element inside the wrapper is clicked', () => {
+      const wrapper = document.createElement('div');
+      const dialog = document.createElement('div');
+      const dialogState = { wrapperId: wrapper.id, wrapper, hide: sinon.spy() };
+
+      wrapper.appendChild(dialog);
+      document.body.appendChild(wrapper);
+
+      addEventListeners(dialogState);
+      dialog.click();
+      expect(dialogState.hide).to.not.have.been.called();
+
+      // Cleanup
+      wrapper.remove();
+    });
+
+    it('executes next() when a next button found via attribute is clicked', () => {
+      const wrapper = document.createElement('div');
+      wrapper.id = 'test';
+      const button = document.createElement('button');
+      button.setAttribute('data-htz-dialog-next', '');
+      wrapper.appendChild(button);
+      document.body.appendChild(wrapper);
+
+      const dialogState = { wrapperId: wrapper.id, wrapper, next: sinon.spy() };
+      addEventListeners(dialogState);
+      button.click();
+      expect(dialogState.next).to.have.been.called();
+
+      // Cleanup
+      wrapper.remove();
+    });
+
+    it('executes prev() when a prev button found via attribute is clicked', () => {
+      const wrapper = document.createElement('div');
+      wrapper.id = 'test';
+      const button = document.createElement('button');
+      button.setAttribute('data-htz-dialog-prev', '');
+      wrapper.appendChild(button);
+      document.body.appendChild(wrapper);
+
+      const dialogState = { wrapperId: wrapper.id, wrapper, prev: sinon.spy() };
+      addEventListeners(dialogState);
+      button.click();
+      expect(dialogState.prev).to.have.been.called();
+
+      // Cleanup
+      wrapper.remove();
+    });
   });
-  it('adds next/prev event listeners to buttons found via attributes');
 
-  it('adds onKeyDown listener to the wrapper for keyCode 27 if element is visible');
+  describe('Keydown events on the wrapper', () => {
+    it('adds keydown listener to the wrapper', () => {
```

Comment:
well, you're right this is brittle. let's remove this test
we can just check behaviour, as I said, strictly speaking this is not completely a unit test, but it does not cross to internal implementation which is better than the current test, 
just perhaps rephrase the test below  to be:
it('makes a visible dialog hidden using hide() when pressing 'esc')